src/topology/sheaves/locally_surjective.lean

```diff
@@ -0,0 +1,162 @@
+@[nolint has_inhabited_instance] -- This may be empty.
+def stalk_set (ℱ : X.presheaf C) (x : X) :=
+  (forget C).obj (ℱ.stalk x)
+local notation `Γₛₜ` : 80 := stalk_set
+
+/-- When (T : ℱ ⟶ 𝒢) is a map of presheaves, we introduce the notation "T _ₛₜ x"
+for the induced map of (underlying objects of) stalks. -/
+def map_on_stalks (T : ℱ ⟶ 𝒢) (x : X) :
+  Γₛₜ ℱ x ⟶ Γₛₜ 𝒢 x := (forget C).map ((Top.presheaf.stalk_functor C x).map T)
+local infix `_ₛₜ` : 80 := map_on_stalks
+
+/-- An equivalent condition for a map of presheaves to be locally surjective
+is for all the induced maps on stalks to be surjective. -/
+def is_surjective_on_stalks (T : ℱ ⟶ 𝒢) :=
+  ∀ (x : X), function.surjective (T _ₛₜ x)
+
+variables [category_theory.limits.preserves_filtered_colimits (forget C)]
+
+/-- Being locally surjective is equivalent to being surjective on stalks. -/
+lemma locally_surjective_iff_surjective_on_stalks (T : ℱ ⟶ 𝒢) :
+  is_locally_surjective T ↔ is_surjective_on_stalks T :=
+begin
+  split; intro hT,
+  { /- human proof:
+    Let g ∈ Γₛₜ 𝒢 x be a germ. Represent it on an open set U ⊆ X
+    as ⟨t, U⟩. By local surjectivity, pass to a smaller open set V
+    on which there exists s ∈ Γ_ ℱ V mapping to t |_ V.
+    Then the germ of s maps to g -/
+
+    -- Let g ∈ Γₛₜ 𝒢 x be a germ.
+    intros x g,
+    -- Represent it on an open set U ⊆ X as ⟨t, U⟩.
+    rcases 𝒢.germ_exist x g with ⟨U, hxU, t, rfl⟩,
+    -- By local surjectivity, pass to a smaller open set V
+    -- on which there exists s ∈ Γ_ ℱ V mapping to t |_ V.
+    rcases hT U t x hxU with ⟨V, ι, hxV, s, h_eq⟩,
+
+
+    -- Then the germ of s maps to g.
+    use (forget C).map (ℱ.germ ⟨x, hxV⟩) s,
+    convert Top.presheaf.stalk_functor_map_germ_apply V ⟨x, hxV⟩ T s,
+
+    -- I said, "then the germ of maps to g."
+    change ((forget C).map _ s) = (forget C).map _ t at h_eq,
+
+    -- The germ of s maps to g? Please?
+    change _ = (forget C).map _ ((forget C).map _ _),
+    rw h_eq,
+
+    -- :(
+    change _ = ((forget C).map _ ≫ (forget C).map _) _,
+    rw [← (forget C).map_comp, 𝒢.germ_res],
+    refl, },
+
+  { /- human proof:
+    Let U be an open set, t ∈ Γ ℱ U a section, x ∈ U a point.
+    By surjectivity on stalks, the germ of t is the image of
+    some germ f ∈ Γₛₜ ℱ x. Represent f on some open set V ⊆ X as ⟨s, V⟩.
+    Then there is some possibly smaller open set x ∈ W ⊆ V ∩ U on which
+    we have T(s) |_ W = t |_ W. -/
+    intros U t x hxU,
+    set t_x := (forget C).map (𝒢.germ ⟨x, hxU⟩) t with ht_x,
+    obtain ⟨s_x, hs_x : (T _ₛₜ x) s_x = t_x⟩ := hT x t_x,
+    obtain ⟨V, hxV, s, rfl⟩ := ℱ.germ_exist x s_x,
+    -- rfl : ℱ.germ x s = s_x
+    have key_W := 𝒢.germ_eq x hxV hxU (T _* s) t
+      (by { convert hs_x,
+            symmetry,
+            convert Top.presheaf.stalk_functor_map_germ_apply _ _ _ s, }),
+    obtain ⟨W, hxW, hWV, hWU, h_eq⟩ := key_W,
+    refine ⟨W, hWU, hxW, ⟨s |_ hWV, _⟩⟩,
+    convert h_eq,
+
+    -- horrible wails from beyond the stars
```

Comment:
Is there an easy way to finish the proof here without the complicated `change`? It should just involve `rw` using `(forget C).map_comp` and `T.naturality`.